modules/mcuboot/hooks/nrf53_hooks.c

```diff
@@ -19,9 +19,56 @@
 #define NET_CORE_VIRTUAL_PRIMARY_SLOT 3
 
 #include <dfu/pcd.h>
+#if defined(CONFIG_PCD_APP) && defined(CONFIG_NRF53_MULTI_IMAGE_UPDATE)
+#include <fw_info_bare.h>
 
-int boot_read_image_header_hook(int img_index, int slot,
-		struct image_header *img_head)
+/** @brief Compare image version numbers of network core and network core update.
+ *
```

Comment:
I'm not seeing any changes to `boot_serial_uploaded_hook` here so someone can completely bypass this just by doing serial recovery? Or is that expected, as an emergency recovery?

Reply:
I would assume that's expected as an emergency recovery but I'll ask the team needing this functionality.

Reply:
No serial recovery does not work like that, if you have downgrade protection enabled, you can load an older image via serial recovery but it will not boot, it needs to be the same or newer than the existing image

Reply:
> No serial recovery does not work like that, if you have downgrade protection enabled, you can load an older image via serial recovery but it will not boot, it needs to be the same or newer than the existing image 

Isn't this statement true only if you have enabled HW downgrade prevention? I can't find any references to downgrade prevention here https://github.com/nrfconnect/sdk-mcuboot/blob/main/boot/boot_serial/src/boot_serial.c 

and MCUBoot only stores the version number in the MCUBoot header AFAIK. So if you use software downgrade prevention Serial Recovery does boot the new image.

Reply:
If you use software protection yes, it is hardware protection that enforces it, but isn't that kind of the point on the nrf5340, because it has hardware protection? Seems odd to have hardware protection on one core and no protection on the other

Reply:
That's how nRF53 hardware is, at least, from my point of view. 

Hardware protection in nRF53 is done using security counters based on OTP. These are in the application core stored in UICR see https://infocenter.nordicsemi.com/topic/ps_nrf5340/chapters/uicr/doc/uicr.html?cp=4_0_0_4_3_2

They are, however, not present in the network core, meaning you cannot have hardware protection in the network core.

https://infocenter.nordicsemi.com/topic/ps_nrf5340/chapters/uicr.network/doc/uicr.network.html?cp=4_0_0_5_3_1